model/project_vars.go

```diff
@@ -441,46 +466,123 @@ func (projectVars *ProjectVars) MergeWithRepoVars(repoVars *ProjectVars) {
 	}
 }
 
-// validParamName is a regexp representing the valid characters for a parameter
-// name. Valid characters are alphanumerics, underscores, dashes, and periods.
-var validParamName = regexp.MustCompile(`^[a-zA-Z0-9_.-]+$`)
-
-// getParamNameForVar returns the corresponding parameter name for a project
-// variable. If the project variable does not yet have a parameter name, it
-// generates one.
-func getParamNameForVar(varsToParams []ParameterMapping, varName string) (string, error) {
-	for _, varToParam := range varsToParams {
-		if varToParam.Name == varName {
-			if varToParam.ParameterName != "" {
-				return varToParam.ParameterName, nil
-			}
-			break
+// convertVarToParam converts a project variable to its equivalent parameter
+// name and value. In particular, it validates that the variable name and value
+// fits within parameter constraints and if the name or value doesn't fit in the
+// constraints, it attempts to fix minor issues where possible. The return value
+// is a valid parameter name and parameter value.
```

Comment:
Same comment here, can we add a todo for the future use of this?

Reply:
I plan to use it in the next sprint for DEVPROD-9405 and DEVPROD-11973, so I think it's okay if it's just unused here for the moment.